pandas/_libs/tslibs/src/datetime/pd_datetime.h

```diff
@@ -0,0 +1,152 @@
+/*
+
+Copyright (c) 2016, PyData Development Team
+All rights reserved.
+
+Distributed under the terms of the BSD Simplified License.
+
+*/
+
+#ifndef PANDAS__LIBS_TSLIBS_SRC_DATETIME_PD_DATETIME_H_
+#define PANDAS__LIBS_TSLIBS_SRC_DATETIME_PD_DATETIME_H_
+
+#ifndef NPY_NO_DEPRECATED_API
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+#endif  // NPY_NO_DEPRECATED_API
+
+#include <numpy/ndarraytypes.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct {
+  npy_int64 days;
+  npy_int32 hrs, min, sec, ms, us, ns, seconds, microseconds, nanoseconds;
+} pandas_timedeltastruct;
+
+const npy_datetimestruct _AS_MIN_DTS = {1969, 12,     31,     23,    59,
```

Comment:
These were originally declared with `extern` linkage and defined within the source file, which was included during compilation of each additional library. I think this was a bit heavy handed, so went with a direct definition here in the header with this refactor